pkg/scheduler/framework/runtime/framework.go

```diff
@@ -153,9 +153,9 @@ func (frw *frameworkImpl) RunScorePlugins(
 	return pluginToClusterScores, nil
 }
 
-func (frw *frameworkImpl) runScorePlugin(ctx context.Context, pl framework.ScorePlugin, spec *workv1alpha2.ResourceBindingSpec, cluster *clusterv1alpha1.Cluster) (int64, *framework.Result) {
+func (frw *frameworkImpl) runScorePlugin(ctx context.Context, pl framework.ScorePlugin, spec *workv1alpha2.ResourceBindingSpec, clusters []*clusterv1alpha1.Cluster, name string) (int64, *framework.Result) {
```

Comment:
Well, it seems like we need pass all cluster objects to adapt this new plugin, but I don't think it's a good idea to change this interface.

If we take a look at the Score plugin interface of kube-scheduler, you will score function only focuses one node. It's a little bit weird when we pass all clusters to a plugin every time.

I understand that you need to know all available replicas and cache them together. So I prefer to add a new scheduler extension point `PreScore` to let you cache all available replicas.

Reply:
Given this case, I guess if we need to optimize our framework first and add some new extension point and cycle state for all plugins to share data.

Reply:
> I understand that you need to know all available replicas and cache them together.

yes.

Reply:
> I guess if we need to optimize our framework first and add some new extension point and cycle state for all plugins to share data.

+1